File: capy/src/e2e-tests/java/dev/capylang/test/OperatorMethodCollisionTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class OperatorMethodCollisionTest {
    @Test
    void shouldKeepNamedAndOperatorMethodsDistinct() {
        var counter = new OperatorMethodCollision.Counter(10);
        assertThat(OperatorMethodCollision.namedPlus(counter, 5)).isEqualTo(15);
        assertThat(OperatorMethodCollision.operatorPlus(counter, 5)).isEqualTo(115);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/OperatorMethodCollision.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('OperatorMethodCollision', () => {
    const collision = generatedModule('dev/capylang/test/OperatorMethodCollision.js');
    const counter = new collision.Counter({ value: 10 });
    assert.equal(collision.namedPlus(counter, 5), 15);
    assert.equal(collision.operatorPlus(counter, 5), 115);
});
